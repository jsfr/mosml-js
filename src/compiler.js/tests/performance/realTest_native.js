"use strict"
var ctimeBegin$1 = function($_var1) {
    return console.time($_var1);
};
var ctimeEnd$2 = function($_var5) {
    return console.timeEnd($_var5);
};
var repeat$3 = function($_var9) {
    return function($_var10) {
        return function($_var11) {
            var $_var13 = function($_var14) {
                return (function() {
                    switch ($_var14) {
                        case 0:
                            return $_var10($_var11)
                        default:
                            return ($_var10($_var11), $_var13($_var14 - 1))
                    }
                }());
            };
            return $_var13($_var9 - 1);
        };
    };
};
var timerep$4 = function($_var35) {
    return function($_var36) {
        return function($_var37) {
            return repeat$3($_var35)(function($_var41) {
                return (ctimeBegin$1($_var37), ($_var36($_var41), ctimeEnd$2($_var37)));
            });
        };
    };
};
var realAdd = function($_var54) {
    var i = 0;
    while (i++ < $_var54) {
        33.3334+9.992;
    }
};
var realSub = function($_var73) {
    var i = 0;
    while (i++ < $_var73) {
        33.3334-9.992;
    }
};
var it$7 = timerep$4(10)(realAdd)("realAdd_native")(10000000.0);
var it$8 = timerep$4(10)(realSub)("realSub_native")(10000000.0);


var realSubTest0 = function($_var73) {
    var i = 0;
    (function() {
        while (i++ < $_var73) {
            33.3334-9.992;
        }
    }())
};
var it$9 = timerep$4(10)(recSubTest)("realSubTest0")(10000000.0);

var realSubTest1 = function($_var73) {
    var i = 0;
    while (i++ < $_var73) {
        (function(){33.3334-9.992}())
    }
};
var it$9 = timerep$4(10)(recSubTest1)("realSubTest1")(10000000.0);

var realSubTest2 = function($_var73) {
    var i = 0;
    (function() {
        while (i++ < $_var73) {
            (function(){33.3334-9.992}())
        }
    }())
};
var it$9 = timerep$4(10)(recSubTest2)("realSubTest2")(10000000.0);






