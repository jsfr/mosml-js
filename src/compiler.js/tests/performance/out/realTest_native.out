==============================JS BEGIN==============================
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 1ms
realAdd_native: 2ms
realAdd_native: 2ms
realAdd_native: 2ms
realAdd_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 1ms
realSub_native: 3ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 2ms
realSub_native: 1ms
realSub_native: 2ms
realSubInnerScope_native: 4ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 4ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 4ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 4ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 2ms
realSubInnerScope_native: 3ms
realSubInnerScope_native: 2ms
realSubOuterScope_native: 16ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 17ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 16ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 16ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 16ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 16ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 16ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 16ms
realSubOuterScope_native: 14ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubOuterScope_native: 15ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 20ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 19ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 19ms
realSubBothScopes_native: 22ms
realSubBothScopes_native: 21ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 15ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 16ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 19ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 17ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 18ms
realSubBothScopes_native: 17ms
==============================JS END================================
